### DISCUSSION TITLE
Iterating for increased accuracy?

### DISCUSSION THREAD

**Discussion by natv** [danielmiessler/Fabric#117] [Ideas]
This is more of an OpenAI issue, but each time a pattern is run (for example, when asked to summarize something), it often misses some detail on each run, yet that detail(s) is included in other runs.

So I was thinking it would be nice if there was a way (when needed) to have it iterate the pattern 2-3 times (produce 3 drafts), and then compare the drafts and combine them into one final version.

I tried to do this using a pattern and instructing it to create drafts and then compare/combine them but it's not working as intended, and possibly just not something that can be done in one instruction set.

So that being said, a cool idea might be to incorporate a new switch, like "--drafts 3" that could so something like this.

**Comment by waldo323** [comment]
Could a separate pattern help implement this?I'm thinking having a separate pattern or multiple patterns would make sense with the 'unix' idea of having small useful parts which work well with each other. in this case we're looking for patterns which aid in repetition and combination for increased quality and coverage.
I think this fits with this project's concept of stitches.

**Comment by chrissonntag** [comment]
An implementation of agents would help. Those would have to be instructed appropriately so that they tried to give the best outcome together. More than less that's exactly what would be needed. There are already some projects that could be used for that: AutoGen, crewAI, Langgraph, etc.
I'm not a good instructor but am able to integrate such a workflow. I will discuss it with Daniel and the others.

**Comment by danielmiessler** [comment]
I'm looking VERY seriously at agent functionality. I'm hoping to have the first version within the next week or two.

Basically, call a pattern, and it does a full install and execution of an agent group doing the task, and then you get the output just like any other pattern.
